src/sentry/api/serializers/models/release.py

```diff
@@ -62,17 +62,19 @@ def get_users_for_authors(organization_id, authors, user=None):
     }
     """
     # Filter users based on the emails provided in the commits
-    user_emails = list(
-        UserEmail.objects.filter(in_iexact("email", [a.email for a in authors])).order_by("id")
+    user_emails = sorted(
+        UserEmail.objects.get_many_from_cache([a.email.lower() for a in authors], key="email"),
```

Comment:
This is sketchy because it's not using `in_iexact` anymore. Going to look at this more.

Reply:
For this to work you need to start caching this by email as well. On the `UserEmail` class, you need to do something like `    objects = BaseManager(cache_fields=("email",))`. Not sure if you can use `key="email__iexact"` there, might be worth testing?